test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -245,7 +245,19 @@ public static RequestOptions expectVersionSpecificWarnings(Consumer<VersionSensi
         expectationsSetter.accept(warningsHandler);
         builder.setWarningsHandler(warningsHandler);
         return builder.build();
-    }    
+    }
+
+    /**
+     * Creates request options designed to be used when calling a deprecated 'typed' API. The
+     * options will ensure that the given warnings are returned if all nodes are on 7.x, and will
+     * allow (but not require) the warnings if any node is running 6.x.
+     *
+     * @param warnings The expected types removal warnings.
+     * @return A {@link RequestOptions} instance for use with a deprecated 'typed' request.
+     */
+    public static RequestOptions expectTypesWarnings(String... warnings) {
```

Comment:
This function looks to be general and not specifically for types. I wonder if it is worth calling it something like `requireWarningsNodesSameVersion`?

Reply:
Technically speaking it would be `requireWarningsIfNodesSameVersionAndAllowWarningsIfMixedVersions`
That's obviously a bit of a mouthful so maybe just `expectWarnings`?

Reply:
👍 I'll update to `expectWarnings` and generalize the javadoc.